plugins/roll.py

```diff
@@ -33,8 +33,40 @@ def parse_dice_syntax(inputs: List[str]) -> List[DieRoll]:
     return [parse_single_die_syntax(expr) for expr in inputs]
 
 
+def mean(sides: int, dice: int) -> float:
+    # https://boardgamegeek.com/blogpost/25470/variance-dice-sums
+    #
+    # example - average result of 2 6-sided dice:
+    #   (2 * 6 + 2) / 2
+    #   (12 + 2) / 2
+    #   14 / 2
+    #   7
+    return (sides * dice + dice) / 2.0
+
+
+def variance(sides: int, dice: int) -> float:
+    # Calculate the variance of once dice of `sides` first
+    # Proof in roll.maxima: maxima < roll.maxima
+    one_dice_variance = (sides**3 - sides) / 72.0
+    return dice * one_dice_variance
+
+
+def clamp(num: int, min_value: int, max_value: int) -> int:
+    return max(min(num, max_value), min_value)
+
+
+def probability_roll(sides: int, dice: int) -> int:
+    m = mean(sides, dice)
+    v = variance(sides, dice)
+    standard_deviation = sqrt(v)
+    roll = normalvariate(m, standard_deviation)
+    return clamp(round(roll), dice, dice * sides)
+
+
 def calculate_roll(roll: DieRoll) -> int:
     total = 0
+    if roll.count > 10000:
+        return probability_roll(roll.die, roll.count) + roll.modifier
```

Comment:
we want to append to the result list rather than return immediately, as you can pass in multiple rolls.

And can we set some flag and prepend a ridiculous message to indicate that MATH is being used? I'm open to suggestions, but the first thing that comes to mind is "soandso rolled 555 USING MATH"

Reply:
So, the current function returns an int, not a list[int].
As for the flag, yeah, that should probably exist.

Reply:
The list is created in the `RollPlugin` class.  That still works as intended.
"USING MATH" is now appended to the result of each roll (when using the probability math stuff).

Reply:
sweet